src/status_im/contexts/settings/wallet/events.cljs

```diff
@@ -95,20 +96,34 @@
   [{:keys [db]} [{:keys [sha3-pwd keypairs-key-uids connection-string on-success]}]]
   (let [key-uid (get-in db [:profile/profile :key-uid])]
     {:fx [[:effects.syncing/import-keypairs-keystores
-           {:key-uid           key-uid
-            :sha3-pwd          sha3-pwd
+           {:key-uid key-uid
+            :sha3-pwd sha3-pwd
             :keypairs-key-uids keypairs-key-uids
             :connection-string connection-string
-            :on-success        (fn [key-uids]
-                                 (rf/call-continuation on-success)
-                                 (rf/dispatch [:wallet/make-keypairs-accounts-fully-operable key-uids]))
-            :on-fail           (fn [error]
-                                 (log/error "failed to import missing key pairs with connection string"
-                                            {:error error})
-                                 (rf/dispatch [:toasts/upsert
-                                               {:type  :negative
-                                                :theme :dark
-                                                :text  (i18n/label :t/incorrect-qr-code)}]))}]]}))
+            :on-success (fn [key-uids]
+                          (rf/call-continuation on-success)
+                          (rf/dispatch [:wallet/make-keypairs-accounts-fully-operable key-uids]))
+            :on-fail (fn [error]
+                       (let [error-message (-> error ex-data :error)
+                             incorrect-keypair?
+                             (string/includes?
+                              error-message
+                              "one or more expected keystore files are not found among the sent files")
```

Comment:
We may be able to handle this error string inside the effect `:effects.syncing/import-keypairs-keystores`, and then that effect could throw a more descriptive exception. Thoughts?

Reply:
I believe it's not needed as we would be displaying a different message for importing a single key pair and for importing all key pairs separately. 

I'm thinking of creating a separate event (as we access app-db for key pair name) to handle the **on-fail** for any cases/flows and keep `wallet/connection-string-for-import-keypair` simple. I haven't created it yet as I need to check the behaviour for all key pairs flow and we can do a clean-up later.